### PR TITLE
Proto for asset registration 

### PR DESCRIPTION
**Pull request by ritwikc**
Changed the protos in pkg/connectors/ to take care of asset registrations.

### CODE REVIEW

File: pkg/connectors/protos/register_asset_request.proto [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+import "credentials.proto";
+import "data_store.proto";
+
+message RegisterAssetRequest {
+    string dataset_id = 1;           // identifier of the bucket created for registering asset
+	Credentials creds = 2;
+	DataStore data_store = 3; 	
+	string destination_catalog_id = 4;
+}
```
Comment (simanadler) [inline pkg/connectors/protos/register_asset_request.proto:18]: Where is the metadata associated with the new data set?  For example, when ingesting the data owner may provide information such as the data residency, level of confidentiality.

File: pkg/connectors/protos/register_asset_request.proto [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+package connectors;
+option java_package = "com.datmesh";
+option go_package = "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf";
+
+import "credentials.proto";
+import "data_store.proto";
+
+message RegisterAssetRequest {
+    string dataset_id = 1;           // identifier of the bucket created for registering asset
```
Comment (simanadler) [inline pkg/connectors/protos/register_asset_request.proto:14]: I'm not clear on what this is.  The comment says it's the bucket, but isn't the bucket in the DataStore?  The param name is dataset_id, but isn't the dataset_id what is returned after the data set is registered in the catalog?

File: connectors/opa/lib/evaluate_opa.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func EvaluatePoliciesOnInput(inputMap map[string]interface{}, opaServerURL strin
 	log.Println("opaServerURL")
 	log.Println(opaServerURL)
 
-	res := performHTTPReq(standardClient, opaServerURL+"v1/data/user_policies", httpMethod, inputJSON, contentType)
+	res := performHTTPReq(standardClient, opaServerURL+"v1/data/wkcpolicy", httpMethod, inputJSON, contentType)
```
Comment (simanadler) [inline connectors/opa/lib/evaluate_opa.go:69]: Why do we have something called wkcpolicy in opa?

**Review by shlomitk1 — CHANGES_REQUESTED** [review]
Many unrelated files - base the branch on master

File: third_party/opa/data-and-policies/wkcpolicy.rego [hunk 4]
```diff
@@ -0,0 +1,75 @@
+package wkcpolicy
```
Comment (simanadler) [inline third_party/opa/data-and-policies/wkcpolicy.rego:1]: We shouldn't have anything wkc specific in the open source.  I'm guessing this is just a naming issue

File: third_party/opa/data-and-policies/wkcpolicy2.rego [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package wkcpolicy
```
Comment (simanadler) [inline third_party/opa/data-and-policies/wkcpolicy2.rego:1]: Same as previous comment

File: pkg/connectors/protos/data_store.proto [hunk 6]
```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package connectors;
+option java_package = "com.datmesh";
+
```
Comment (shlomitk1) [inline pkg/connectors/protos/data_store.proto:8]: go_package is missing

**Comment by ritwikc** [comment]
Closing this PR as I have created a new PR #330 for the same.

**Pull request closed without merging**